File: java/src/main/java/ai/buble/sdk/chat/ChatModelsService.java
package ai.buble.sdk.chat;

import ai.buble.sdk.http.BubleHttpClient;
import com.fasterxml.jackson.core.type.TypeReference;

/**
 * Chat model discovery methods.
 */
public final class ChatModelsService {
    private static final TypeReference<ChatModelList> CHAT_MODEL_LIST =
            new TypeReference<ChatModelList>() {};

    private final BubleHttpClient http;

    public ChatModelsService(BubleHttpClient http) {
        this.http = http;
    }

    public ChatModelList list() {
        return http.get("/api/v1/models", CHAT_MODEL_LIST);
    }
}

File: dotnet/src/Buble.Sdk/Chat/ChatModelsService.cs
using Buble.Sdk.Http;

namespace Buble.Sdk.Chat;

public sealed class ChatModelsService
{
    private readonly BubleHttpClient _http;

    internal ChatModelsService(BubleHttpClient http)
    {
        _http = http;
    }

    public Task<ChatModelList?> ListAsync(CancellationToken cancellationToken = default)
    {
        return _http.GetAsync<ChatModelList>("/api/v1/models", cancellationToken: cancellationToken);
    }
}
